types/field_type.go

```diff
@@ -111,13 +111,33 @@ func (ft *FieldType) Init(tp byte) {
 	ft.Decimal = UnspecifiedLength
 }
 
+// GetDefaultFieldLengthAndDecimal returns the default display length (flen)
+// and decimal length for column,  taking into account the sign.
+// Refer: source of MySQL 5.7, adjust_max_effective_column_length()
+func (ft *FieldType) GetDefaultFieldLengthAndDecimal() (flen int, decimal int) {
+	flen, decimal = mysql.GetDefaultFieldLengthAndDecimal(ft.Tp)
```

Comment:
Why not directly modify `mysql.GetDefaultFieldLengthAndDecimal()` to consider the unsigned flag?